Node.js/sampleresponse.js

```diff
@@ -40,14 +40,16 @@ var mqBoilerPlate = new MQBoilerPlate();
 
 async function msgCB(md, buf) {
   debug_info('Message Received');
-  var ok = true
+  let ok = true;
   if (md.Format == "MQSTR") {
-    var msgObject = null;
+    let msgObject = null;
     try {
       msgObject = JSON.parse(buf);
-      debug_info("JSON Message Object found", msgObject);
-      ok= await respondToRequest(msgObject, md);
-     
+      debug_info('JSON Message Object found', msgObject);
+      if (ok) {
+        debug_info('Starting response sequence');
+        ok = await respondToRequest(msgObject, md);
+      }
```

Comment:
This function is always returning true. ie. the async promise is always resolving to true. It needs to return `ok`, so that its promise can resolve to false.